### PR TITLE
[O2B-898] Clean lost runs and environments every 30 seconds

### PR DESCRIPTION
**Pull request by martinboulais**
#### I have a JIRA ticket
- [X] branch and/or PR name(s) include(s) JIRA ID
- [X] issue has "Fix version" assigned
- [X] issue "Status" is set to "In review"
- [X] PR labels are selected

Notable changes for developers:
- A new route has been added to purge lost runs and detectors

### CODE REVIEW

File: lib/application.js [hunk 1]
```diff
@@ -54,6 +55,16 @@ class BookkeepingApplication {
...
+
+            // Start background housekeeping
+            setInterval(async () => {
+                try {
+                    await handleLostRunsAndEnvironments();
+                    this.logger.debug('Updated lost runs and environments');
+                } catch (error) {
+                    this.logger.error(`Error while handling lost logs: ${error}`);
+                }
+            }, 15 * 60 * 1000);
```
Comment (graduta) [inline lib/application.js:67]: I would say 15 minutes will for sure impact the statistics.
Initially, we agree on 15 seconds but to be on the safe side, perhaps we start with 30 seconds?

File: lib/server/controllers/environments.controller.js
Comment (graduta) [inline lib/server/controllers/environments.controller.js:1]: As we are not using a ROUTE anymore, is there a point of these imports in the controller?

File: lib/server/services/environment/transitionToErrorLostEnvironments.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ *
+ * @param {string[]} activeEnvironmentIds the ids of active environments (those environments will not be transitioned to error)
+ * @param {number} modificationTimeWindow the time we have (in MINUTES), after the creation of an environment, to purge them. If an environment
+ *     has been created more than this amount of minutes before NOW, it is NOT purged
+ * @return {Promise<void>} resolve once all inactive environments has been transitioned to error
+ * @deprecated
+ */
+exports.transitionToErrorLostEnvironments = async (activeEnvironmentIds, modificationTimeWindow) => sequelize.query(`
+    INSERT INTO environments_history_items (environment_id, status, status_message)
+    SELECT e.id, 'ERROR', 'Shutdown not received, environment not existing anymore'
```
Comment (graduta) [inline lib/server/services/environment/transitionToErrorLostEnvironments.js:27]: I would rephrase it a bit to something more like: `Shutdown not received but environment does not exist anymore`

**Review by graduta — CHANGES_REQUESTED** [review]
As the PR feature changed to not use an API endpoint anymore, please update the PR title as well.

File: lib/application.js (hunk 1 above)
Comment (martinboulais) [reply lib/application.js:67]: I misunderstood, I also thought it was a lot... Let's try with 30 seconds yes

File: lib/server/controllers/environments.controller.js
Comment (martinboulais) [reply lib/server/controllers/environments.controller.js:1]: No indeed, as the linter says

File: lib/server/services/environment/transitionToErrorLostEnvironments.js (hunk 2 above)
Comment (martinboulais) [reply lib/server/services/environment/transitionToErrorLostEnvironments.js:27]: Done

File: lib/application.js [hunk 3]
```diff
@@ -54,6 +55,16 @@ class BookkeepingApplication {
...
                 await this.gRPCServer.listen(gRPCOrigin);
             }
+
+            // Start background housekeeping
+            setInterval(async () => {
+                try {
+                    await handleLostRunsAndEnvironments();
+                    this.logger.debug('Updated lost runs and environments');
+                } catch (error) {
+                    this.logger.error(`Error while handling lost logs: ${error}`);
```
Comment (graduta) [inline lib/application.js:65]: I believe it is not the `logs` that we are handling here

**Review by graduta — APPROVED** [review]
